SimTKcommon/Simulation/include/SimTKcommon/internal/StateImpl.h

```diff
@@ -636,7 +636,7 @@ class SimTK_SimTKCOMMON_EXPORT StateImpl {
 
     StateImpl& operator=(const StateImpl& src);
 
-    ~StateImpl() {}   // default destructor
+    ~StateImpl() {delete stateLock;}
 
     // Copies all the variables but not the cache.
     StateImpl* clone() const {return new StateImpl(*this);}
```

Comment:
This doesn't seem right. Won't this copy the mutex pointer meaning every copy of this State will share the same mutex? What is supposed to happen to the mutex when a state is:
- copy constructed
- copy assigned
- move constructed
- move assigned